File: src/discount.py
def main():
    calc = lambda pricing, discount: pricing * (100 - discount) / 100

    try:
        price = int(input('Enter the product price $: '))
        per = int(input('What is your discount? %: '))

        print(f'Total price to pay: ${calc(price, per)}')
    except ValueError:
        print('Only can introduce numbers \n')
        main()

if __name__ == '__main__':
    main()

File: src/discount.rb
# Lambda use in a simple Ruby code


calc = lambda { |pricing, discount| pricing * (100 - discount) / 100 }

puts "Enter the product price: "
price = gets.chomp.to_i

puts "Enter your discount %: "
percent = gets.chomp.to_i

puts "Total price to pay $#{calc.call(price, percent)}"